File: C/Assignments/Assignmrnt 14/salesmanager.c
#include <stdio.h>
#include <string.h>

struct SalesManager {
    int id, target;
    char name[50];
    float salary, incentive;
};

void storeByValue(struct SalesManager sm);
void displayByValue(struct SalesManager sm);

void storeByAddress(struct SalesManager *sm);
void displayByAddress(struct SalesManager *sm);

int main() {
    struct SalesManager sm1, sm2, arr[2];

    storeByValue(sm1);
    displayByValue(sm1);

    storeByAddress(&sm2);
    displayByAddress(&sm2);

    for(int i=0;i<2;i++){
        printf("\nEnter details for SalesManager %d:\n", i+1);
        storeByAddress(&arr[i]);
    }
    for(int i=0;i<2;i++){
        printf("\nDetails of SalesManager %d:\n", i+1);
        displayByAddress(&arr[i]);
    }
    return 0;
}

void storeByValue(struct SalesManager sm){
    printf("Enter ID: "); scanf("%d", &sm.id);
    printf("Enter Name: "); scanf("%s", sm.name);
    printf("Enter Salary: "); scanf("%f", &sm.salary);
    printf("Enter Incentive: "); scanf("%f", &sm.incentive);
    printf("Enter Target: "); scanf("%d", &sm.target);
}

void displayByValue(struct SalesManager sm){
    printf("\nID: %d", sm.id);
    printf("\nName: %s", sm.name);
    printf("\nSalary: %.2f", sm.salary);
    printf("\nIncentive: %.2f", sm.incentive);
    printf("\nTarget: %d\n", sm.target);
}

void storeByAddress(struct SalesManager *sm){
    printf("Enter ID: "); scanf("%d", &sm->id);
    printf("Enter Name: "); scanf("%s", sm->name);
    printf("Enter Salary: "); scanf("%f", &sm->salary);
    printf("Enter Incentive: "); scanf("%f", &sm->incentive);
    printf("Enter Target: "); scanf("%d", &sm->target);
}

void displayByAddress(struct SalesManager *sm){
    printf("\nID: %d", sm->id);
    printf("\nName: %s", sm->name);
    printf("\nSalary: %.2f", sm->salary);
    printf("\nIncentive: %.2f", sm->incentive);
    printf("\nTarget: %d\n", sm->target);
}

File: Java/Assignment/EmployeeMiniProjWithFileHandling/src/Model/SalesManager.java
package Model;

import Model.Employee;

public class SalesManager extends Employee {

	int target;
	double incentive;
	
	public SalesManager() {
		this.target = 0;
		this.incentive = 0;
	}
	
	public SalesManager(int id, String name, double salary,int target, double incentive) {
		super(id,name,salary);
		this.target = target;
		this.incentive = incentive;
	}

	public int getTarget() {
		return target;
	}

	public void setTarget(int target) {
		this.target = target;
	}

	public double getIncentive() {
		return incentive;
	}

	public void setIncentive(double incentive) {
		this.incentive = incentive;
	}

	public String toString() {
		return super.toString()+"\ntarget:" + target + "\nincentive:" + incentive+"\n";
	}

	public double calculateSalary() {
		return this.incentive+this.getSalary();
	}

}
